src/panels/ItemPanel.tsx

```diff
@@ -174,37 +154,28 @@ export default class ItemPanel extends React.Component<Props, State> {
 								modalRemoveItem: true
 							})
 						}}
-						checkboxChecked={this.state.selectedItems.includes(id)}
-						handleShowCheckbox={() => {
-							let items = this.state.selectedItems
-							if (items.includes(id)) {
-								items.splice(items.indexOf(id), 1)
-							} else {
-								items.push(id);
-							}
-							this.setState({selectedItems: items, selectionMode: (items.length > 0)})
+						handleSelect={() => {
+							if (ProjectSettings.selectedElements.includes(id))
```

Comment:
All of those functions within the `<PackageItem>` component definition are created separately for each item and on each render. That's not efficient, but may be good enough for low number of packages.

Reply:
I tried to put the functions into the classes themselves where possible.